### ISSUE TITLE
[Epic] 0.A Package Identity and Runtime Bootstrap

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/pyrox#7]
Source draft ID: `PYX-02`

The wave-level design context is copied below so this issue stays self-contained for another implementing agent.

## Shared Context
This grouped draft is meant to be self-contained enough that another agent can implement the Wave 0 backlog without opening the private design docs. The source material for this wave came from the `README`, `vision`, `architecture`, `boundaries`, `decisions`, `api/core.md`, and `examples/core.md` design docs for `pyrox`.

`pyrox` is a unified package with three top-level areas:

- `pyrox._core`: the Equinox-to-NumPyro bridge.
- `pyrox.gp`: Gaussian process building blocks and protocols.
- `pyrox.nn`: Bayesian and uncertainty-aware neural network layers.

Wave 0 exists because the current repo is still a template repo. The local package is still named `mypackage`, the docs are template docs, and the source tree does not yet express the merged-package identity described in the design docs.

## Design Snapshot
The design docs freeze three important modeling patterns that later waves depend on:

- Pattern A: a module owns its own probabilistic semantics via `PyroModule`, `pyro_param`, and `pyro_sample`.
- Pattern B: a module stays pure Equinox, while the NumPyro model injects sampled values with `eqx.tree_at`.
- Pattern C: `Parameterized` owns constrained parameters, priors, guide metadata, and mode switching for GP-style modules.

Wave 0 only needs to preserve room for those patterns. Their concrete implementation moves into the dedicated Core wave immediately after this scaffold wave.

## Intended Package Layout
Wave 0 should leave the repo shaped for later waves, even if most modules are mostly stubs:

```text
src/pyrox/
  __init__.py
  _core/
    __init__.py
    pyro_module.py
    descriptors.py
    parameterized.py
    utils.py
  gp/
    __init__.py
  nn/
    __init__.py
```

This wave does not need to implement the full internals of every file above. It only needs to make the package identity and import surface line up with that layout so the next wave can implement `_core` for real.

## Runtime Boundary
The architecture docs position these as the core dependency boundaries:

- Required runtime: `jax`, `equinox`, `numpyro`, `gaussx`, `lineax`
- Optional runtime: `optax`
- Example/docs only: `einops`, `matplotlib`, `numpy`, `scikit-learn`

Wave 0 should preserve that distinction instead of flattening everything into one dependency bucket.

---

## Theme
Replace the template package identity with real `pyrox` packaging, metadata, and runtime dependency boundaries.

## Parent Wave
- Wave epic: `#6`
- Wave label: `wave:0`
- Milestone: `v0.0-scaffold`

## Parallelism
- Can run in parallel with: none (single-theme wave)
- Blocked by (inside this wave): none
- Must complete before: #8

## Definition of Done
- Package name, import paths, docs metadata, and repository URLs all point at `pyrox`.
- Required runtime and optional dependency groups are explicit and consistent with the architecture docs.

## Relationships
- Parent wave: #6.
- Blocks the dedicated Core wave where `_core` and quality surfaces become real.

**Comment by jejjohnson** [comment]
All Epic 0.A deliverables shipped in PR #53.
